frontend/packages/console-shared/src/utils/operator-utils.ts

```diff
@@ -0,0 +1,40 @@
+import * as _ from 'lodash';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { ClusterServiceVersionKind } from '@console/operator-lifecycle-manager/src';
+import { CAMEL_SOURCE_INTEGRATION } from '@console/knative-plugin';
+import { TopologyDataResources } from '@console/dev-console/src/components/topology';
+import { getDynamicEventSourcesModelRefs } from '@console/knative-plugin/src/utils/fetch-dynamic-eventsources-utils';
+import { TopologyResources } from '../types';
+import { getOperatorBackedServiceKindMap, getKnativeDynamicResources } from './resource-utils';
+
+export const isOperatorBackedKnResource = (
+  obj: K8sResourceKind,
+  resources: TopologyDataResources,
+) => {
+  const eventSourceProps = getDynamicEventSourcesModelRefs();
+  return !!_.find(getKnativeDynamicResources(resources, eventSourceProps), (evsrc) =>
+    obj.metadata?.labels?.[CAMEL_SOURCE_INTEGRATION]?.startsWith(evsrc.metadata.name),
+  );
+};
+
+export const isOperatorBackedService = (
+  obj: K8sResourceKind,
+  installedOperators: ClusterServiceVersionKind[],
+  resources?: TopologyResources,
+): boolean => {
+  const kind = _.get(obj, 'metadata.ownerReferences[0].kind', null);
+  const ownerUid = _.get(obj, 'metadata.ownerReferences[0].uid');
+  // added this as needs to hide oprator backed if belong to source
+  if (resources && isOperatorBackedKnResource(obj, resources)) {
+    return false;
+  }
```

Comment:
We should not be checking knative edge cases for topology rendering here. If a resource is operator backed then it is.